src/waveform/waveformwidgetfactory.h

```diff
@@ -231,9 +232,16 @@ class WaveformWidgetFactory : public QObject, public Singleton<WaveformWidgetFac
     void renderSelf();
     void swapSelf();
 
+    void addHandle(
+            QHash<WaveformWidgetType::Type, QList<WaveformWidgetBackend>>&
+                    collectedHandles,
```

Comment:
I don't like the out-parameter here, but we can clean that up some time else.

Reply:
I agree, I was thinking to put all this in a class with the QHash as a member. I'll add a TODO so we don't forget about it.

Reply:
Just referencing [mixxxdj/mixxx#13245](https://github.com/mixxxdj/mixxx/issues/13245) for tracking